mu/interface/panes.py

```diff
@@ -498,6 +499,9 @@ def contextMenuEvent(self, event):
         # Mu micro:bit mode only handles .py & .hex
         if ext == ".py" or ext == ".hex":
             open_internal_action = menu.addAction(_("Open in Mu"))
+            write_to_main_action = menu.addAction(
+                _("Write to main.py on device")
+            )
```

Comment:
If this menu item is added for hex files as well, does that mean it will try to transfer the entire hex as a 'main.py' file?

Reply:
Yeah, but that doesn't make any sense. I'll take a look.

Reply:
I guess this only makes sense for the ESP-mode, so I probably need to add in a different way from within the mode, and only for .py files.

Reply:
In principle, when the modes are detached from the editor, this will be easier to do. Right now, we'll risk adding even more spaghetti code if we try to allow modes to add a context menu item.

In the future I hope a mode will be able to define its own UI components, instead of the UI components have to be shared by many modes, e.g. provide a default filesystem pane, and allow modes to inherit and customize it (the same for the plotter, REPL, etc.).

So I think the best we could do now, is to add this for all the modes (but only for .py), instead of introducing tighter couplings between modes and panes as it is.

How does that sound?

Reply:
@carlosperate

Reply:
Yeah, doing it only for .py files for now sounds good.
Technically it could also make sense for the micro:bit mode, as a hex file can contain a Python script (that's how uFlash adds code to a hex right now), but as you mentioned, we could add that once we have more granular control from individual modes.